src/components/MatchRuleList.tsx

```diff
@@ -0,0 +1,255 @@
+    console.log(id)
+    let newMatchRules = [...matchRules]
+    const ruleIndex = matchRules.findIndex(rule => rule.id === id)
+
+    newMatchRules[ruleIndex].toDelete = true
+    console.log(newMatchRules)
+    setMatchRules(newMatchRules)
+  }
+
+  const saveMatchRules = async () => {
+    let updatedMatchRules: MatchRule[] = []
+
+    const requests = matchRules.map((rule, index) => {
+      // If the rule should be deleted, we delete it. This is done before checking validity, because
+      // a rule might have been edited to be invalid and then deleted
+      //
+      // Rules that have not been created yet are also skipped, since they will be gone with the re-render
+      if (rule.toDelete && rule.createdAt) {
+        return matchRuleApi.delete(rule).catch(err => setError(err.message))
+      }
+
+      // If the rule is missing an account ID or a match, it is invalid, so we discard it
+      if (!rule.accountId || !rule.match) {
+        return true
+      }
+
+      if (rule.createdAt) {
+        // createdAt is set by the backend, so this is an existing rule that needs to be updated
+        rule.priority = index
+
+        return matchRuleApi
+          .update(rule)
+          .then(updatedRule => {
+            updatedMatchRules.push(updatedRule)
+          })
+          .catch(err => setError(err.message))
+      } else {
+        // This is a new rule, it needs to be created
+        rule.priority = index
+        return matchRuleApi
+          .create(rule, budget)
+          .then(newRule => {
+            updatedMatchRules.push(newRule)
+          })
+          .catch(err => setError(err.message))
+      }
+    })
+
+    await Promise.all(requests)
+
+    setMatchRules(updatedMatchRules.sort((a, b) => a.priority - b.priority))
+    setNotification(t('changesSaved'))
+  }
+
+  useEffect(() => {
+    if (!isLoading) {
+      setIsLoading(true)
+    }
+
+    const promises = [
+      matchRuleApi.getAll(budget).then(setMatchRules),
+      accountApi.getAll(budget).then(setAccounts),
+    ]
+
+    Promise.all(promises)
+      .then(() => setError(''))
+      .catch(err => {
+        setError(err.message)
+      })
+      .then(() => setIsLoading(false))
+  }, [budget])
+
+  return (
+    <>
+      <div className="header mb-4">
+        <h1>{t('matchRules.matchRules')}</h1>
+
+        <div className="header--action full-centered">
+          <button onClick={() => setHideHelp(!hideHelp)} title={t('help')}>
+            <QuestionMarkCircleIcon className="icon-sm" />
+          </button>
+
+          <button onClick={() => addMatchRule()} title={t('matchRules.add')}>
+            <PlusIcon className="icon-sm" />
+          </button>
+
+          <button onClick={() => saveMatchRules()} title={t('save')}>
+            <CloudArrowUpIcon className="icon-sm" />
+          </button>
+        </div>
+      </div>
+
+      <Error error={error} />
+
+      <div className={`card p-4 md:my-4 ${hideHelp && 'hidden'}`}>
```

Comment:
Why does this only have vertical spacing on big screens? 

```suggestion
      <div className={`card p-4 my-4 ${hideHelp && 'hidden'}`}>
```
It currently looks like this on mobile:
![image](https://github.com/user-attachments/assets/4fb6fe1e-a08a-424b-bed0-cc2311ac2409)

Reply:
That's an oversight, fixed in 35ccdfe7c361213ca7818e8e348eb4a45ec4783a.